sources/HUBUtilities.h

```diff
@@ -95,6 +116,76 @@ static inline NSDictionary<NSString *, id> * _Nullable HUBMergeDictionaries(NSDi
 }
 
 /**
+ *  Return the property names that should be taken into account when handling a `UINavigationItem`
+ *
+ *  This function is used to determine which properties should be compared or copied. Note that
+ *  `HUBAutoEquatable` is not used here, since we don't control the implementation of the class.
+ */
+static inline NSArray<NSString *> *HUBNavigationItemPropertyNames() {
+    return @[
+        HUBKeyPath((UINavigationItem *)nil, title),
+        HUBKeyPath((UINavigationItem *)nil, titleView),
+        HUBKeyPath((UINavigationItem *)nil, prompt),
+        HUBKeyPath((UINavigationItem *)nil, backBarButtonItem),
+        HUBKeyPath((UINavigationItem *)nil, hidesBackButton),
+        HUBKeyPath((UINavigationItem *)nil, leftBarButtonItems),
+        HUBKeyPath((UINavigationItem *)nil, rightBarButtonItems),
+        HUBKeyPath((UINavigationItem *)nil, leftItemsSupplementBackButton)
+    ];
+}
+
+/**
+ *  Return whether two `UINavigationItem` instances are equal
+ *
+ *  @param navigationItemA The first navigation item
+ *  @param navigationItemB The second navigation item
+ *
+ *  The two instances are considered equal if all properties which have names included in the array
+ *  obtained by calling `HUBNavigationItemPropertyNames()` are equal.
+ */
+static inline BOOL HUBNavigationItemEqualToNavigationItem(UINavigationItem *navigationItemA, UINavigationItem *navigationItemB) {
```

Comment:
What’s the reason we need to inline this function? Seems like it could just be a normal function instead, no?

The overhead of executing the function should be very minimal compared to the complexity of it’s implementation. So might as well have it be a normal function imo.